server/bin/pbench-backup-tarballs.py

```diff
@@ -237,12 +236,15 @@ def backup_to_s3(s3_obj, logger, controller_path, controller, tb, tar, resultnam
 
     # Check if the result already present in s3 or not
     try:
-        obj = s3_obj.get_tarball_header(Bucket=s3_obj.bucket_name,
+        tbh = s3_obj.get_tarball_header(Bucket=s3_obj.bucket_name,
                                         Key='{}'.format(s3_resultname))
-    except Exception:
+    except NoSuchKey:
+        s3_md5 = None
+    except Exception as e:
+        logger.error(e)
```

Comment:
This is not a new code path, but it does seem like a logic error.  Why would a general `Exception` cause us to consider the tar ball missing from the S3 instance?  Seems like for this tar ball, we'd only want to report the error talking to S3, and move to the next tar ball to consider.